core/types/l2trace.go

```diff
@@ -166,10 +166,16 @@ func NewTransactionData(tx *Transaction, blockNumber uint64, config *params.Chai
 	signer := MakeSigner(config, big.NewInt(0).SetUint64(blockNumber))
 	from, _ := Sender(signer, tx)
 	v, r, s := tx.RawSignatureValues()
+
+	nonce := tx.Nonce()
+	if tx.IsL1MessageTx() {
+		nonce = tx.L1MessageQueueIndex()
+	}
+
 	result := &TransactionData{
```

Comment:
Regarding the BatcherHeader Encoding in new rollup design, Chunk uses [TransactionData](https://github.com/scroll-tech/scroll/blob/chunk-encoding/common/types/chunk.go#L16), and when we create a [new BatchHeader](https://gist.github.com/Thegaram/2cb0ab03a1c08d1b95ce9e2af5933389#file-rollup_v2-go-L30), we need to somehow obtain `skippedL1MessageBitmap` from `Chunk`. Currently, is there a way to notice if a l1tx is skipped?

Reply:
The version in this PR does not allow skipping. Skipping will be added in https://github.com/scroll-tech/go-ethereum/pull/321.

The only way is to find "gaps" in the `L1MessageTx.QueueIndex` for messages included in a block/chunk/batch. And also between the last L1 message of the parent batch and the first one in the current batch.